doc/introduction/remote.rst

```diff
@@ -352,52 +288,101 @@ Rgate(1.902) | (q[0])
+operations that are applied on the physical chip.
 
->>> from strawberryfields.io import save
->>> save("test_compiled.xbb", prog)
 
+Working with Blackbird scripts
+------------------------------
 
-Tips and tricks
----------------
+When submitting quantum programs to be executed remotely, they are communicated to
+the cloud platform using Blackbird --- a quantum photonic assembly language.
+Strawberry Fields also supports exporting programs directly as Blackbird scripts
+(an ``xbb`` file); Blackbird scripts can then be submitted to be executed via the
+Strawberry Fields :doc:`command line interface </code/sf_cli>`.
 
-.. code-block:: python3
+For example, lets save our program above as a Blackbird script in the
+current directory, via the :func:`~.save` function:
 
-    from strawberryfields.utils import operation
+>>> sf.save("program1.xbb", prog)
 
-We can define an operation to make it easier to apply the same unitary
-to both signal and idler modes.
+This produces the following Blackbird script, where comments have been added
+for clarity:
 
 .. code-block:: python3
 
-    @operation(4)
-    def unitary(q):
-        ops.Interferometer(U) | q
-        ops.BSgate(0.543, 0.123) | (q[2], q[0])
+    name remote_job1
+    version 1.0
+    target X8_01 (shots = 20)
 
-    prog = sf.Program(8)
+    complex array A0[4, 4] =
+        -0.13879438-0.47517904j, -0.29303954-0.47264099j, -0.43951987+0.12977568j, -0.03496718-0.48418713j
+        0.06065372-0.11292765j, 0.54733962+0.1215551j, -0.50721513+0.56195975j, -0.15923161+0.26606674j
+        0.42212573-0.53182417j, -0.2642572+0.50625182j, 0.19448705+0.28321781j, 0.30281396-0.05582391j
+        0.43097587-0.30288974j, 0.07419772-0.21155126j, 0.28335618-0.13633175j, -0.75113453+0.09580304j
 
-    with prog.context as q:
-        ops.S2gate(1.0) | (q[0], q[4])
-        ops.S2gate(1.0) | (q[1], q[5])
-        ops.S2gate(1.0) | (q[2], q[6])
-        ops.S2gate(1.0) | (q[3], q[7])
+    complex array A1[4, 4] =
+        -0.13879438-0.47517904j, -0.29303954-0.47264099j, -0.43951987+0.12977568j, -0.03496718-0.48418713j
+        0.06065372-0.11292765j, 0.54733962+0.1215551j, -0.50721513+0.56195975j, -0.15923161+0.26606674j
+        0.42212573-0.53182417j, -0.2642572+0.50625182j, 0.19448705+0.28321781j, 0.30281396-0.05582391j
+        0.43097587-0.30288974j, 0.07419772-0.21155126j, 0.28335618-0.13633175j, -0.75113453+0.09580304j
+
+    # Initial states are two-mode squeezed states
+    S2gate(1.0, 0.0) | [0, 4]
+    S2gate(1.0, 0.0) | [1, 5]
+    S2gate(1.0, 0.0) | [3, 7]
 
-        unitary() | q[:4]
-        unitary() | q[4:]
+    # Apply the unitary matrix above to
+    # the first pair of modes, as well
+    # as a beamsplitter
+    Interferometer(A0) | [0, 1, 2, 3]
+    BSgate(0.543, 0.123) | [2, 0]
+    Rgate(0.453) | 1
+    MZgate(0.65, -0.54) | [2, 3]
+
+    # Duplicate the above unitary for
+    # the second pair of modes
+    Interferometer(A1) | [4, 5, 6, 7]
```

Comment:
Interesting that it prefers to use another identical matrix `A1` instead of going for `A0` again.

Reply:
Yep, I didn't know this myself until now!